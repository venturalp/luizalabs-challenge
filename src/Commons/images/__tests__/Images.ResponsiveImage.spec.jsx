import { render, waitFor } from '@testing-library/react'
import React from 'react'
import { ResponsiveImage } from '../Images.ResponsiveImage'
import LogoSmall from 'Assets/logo.png'
import Logo from 'Assets/logo2x.png'
import { sleepTest } from 'Commons/tests/Tests.Helpers'

const logoImgQueries = [
  {
    img: LogoSmall,
    size: 0,
  },
  {
    img: Logo,
    size: 700,
  },
]
describe('test responsive image', () => {
  it('test resize window', async () => {
    global.window.innerWidth = 768
    const { asFragment } = render(<ResponsiveImage queries={logoImgQueries} />)

    expect(asFragment()).toMatchSnapshot()

    global.window.innerWidth = 500
    global.window.dispatchEvent(new Event('resize'))
    await sleepTest()
    await waitFor(() => expect(asFragment()).toMatchSnapshot())
  })
})
