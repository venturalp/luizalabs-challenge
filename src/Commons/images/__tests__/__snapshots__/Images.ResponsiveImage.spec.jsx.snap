// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test responsive image test resize window 1`] = `
<DocumentFragment>
  <img
    class="sc-bdfBwQ dJiJvq"
    src="logo2x.png"
  />
</DocumentFragment>
`;

exports[`test responsive image test resize window 2`] = `
<DocumentFragment>
  <img
    class="sc-bdfBwQ dJiJvq"
    src="logo.png"
  />
</DocumentFragment>
`;
